pyensign/ensign.py

```diff
@@ -60,18 +62,41 @@ def __init__(
         connection = Connection(addrport=endpoint, insecure=insecure, auth=auth)
         self.client = Client(connection)
 
-    async def publish(self, events):
+    async def publish(self, topic_name, *events):
         """
-        Publish events to the Ensign server.
+        Publish events to an Ensign topic.
 
         Parameters
         ----------
-        events : iterable of api.v1beta1.event_pb2.Event
+        topic_name : str
+            The name of the topic to publish events to.
+
+        events : iterable of events
             The events to publish.
         """
 
+        if topic_name == "":
+            raise ValueError("topic_name is required")
+
+        if len(events) == 0:
+            raise ValueError("no events provided")
+
+        # Get the topic ID from the topic name
+        topic_id = await self.topic_id(topic_name)
+
+        # Ensure the topic exists
+        if topic_id == "":
+            topic = await self.create_topic(topic_name)
+            topic_id = str(ULID.from_bytes(topic.id))
+
+        # TODO: Support user-defined generators
```

Comment:
If you want, instead of creating a Shortcut story, we could add this to the backlog issues here on Github, so that users can see what we have queued up (and that they could contribute, if they're looking for ideas).

Reply:
I think that's a good idea!